pkg/service/ipam-service.go

```diff
@@ -0,0 +1,204 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bufbuild/connect-go"
+	goipam "github.com/metal-stack/go-ipam"
+	v1 "github.com/metal-stack/go-ipam/api/v1"
+	"go.uber.org/zap"
+	"inet.af/netaddr"
+)
+
+type IPAMService struct {
+	log    *zap.SugaredLogger
+	ipamer goipam.Ipamer
+}
+
+func New(log *zap.SugaredLogger, ipamer goipam.Ipamer) *IPAMService {
+	return &IPAMService{
+		log:    log,
+		ipamer: ipamer,
+	}
+}
+
+func (i *IPAMService) CreatePrefix(_ context.Context, req *connect.Request[v1.CreatePrefixRequest]) (*connect.Response[v1.CreatePrefixResponse], error) {
+	i.log.Debugw("createprefix", "req", req)
+	// FIXME context must be passed here
+	resp, err := i.ipamer.NewPrefix(req.Msg.Cidr)
```

Comment:
Are there any plans on request validation? Would be nice to do it here before separate DB layers start to implement this individually.

Reply:
It is already validated in the impl which is in front of the db layers, see prefix.go

Reply:
done